core/server/worker/src/main/java/alluxio/worker/block/DefaultWorkerMetaStore.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block;
+
+import static alluxio.Constants.CLUSTERID_FILE;
+
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.util.io.FileUtils;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+/**
+ * persistence worker info to a file.
+ */
+@NotThreadSafe
+public class DefaultWorkerMetaStore implements WorkerMetaStore {
+  private static final Logger LOG = LoggerFactory.getLogger(WorkerMetaStore.class);
+  /*Use Properties to persistence info */
+  final Properties mProperties;
+  /*Properties file path */
+  final String mFile;
+
+  DefaultWorkerMetaStore(String path) throws IOException {
+    mProperties = new Properties();
+    mFile = path;
+    initFile(); // If the file does not exist, it will be created
+  }
+
+  DefaultWorkerMetaStore() throws IOException {
+    this(PathUtils.concatPath(
+        ServerConfiguration.get(PropertyKey.WORKER_METASTORE_PATH), CLUSTERID_FILE));
+  }
+
+  @Override
+  public void reset() throws IOException {
+    if (!Files.exists(Paths.get(mFile))) {
+      return;
+    }
+    // clear mProperties in memory
+    mProperties.clear();
+    // clear the persistence file
+    try (OutputStream os = new FileOutputStream(mFile)) {
+      os.write("".getBytes(StandardCharsets.UTF_8));
+    } catch (Exception e) {
+      LOG.error("Failed to resetState file.", e);
+    }
+  }
+
+  /**
+   * Searches for the property with the specified key in file.
+   * If the property file does not exist or key is not found in this property list
+   * will return empty string
+   *
+   * @param key the property key
+   * @return the value in this property list with the specified key value
+   */
+  @Override
+  public String get(String key) {
+    Preconditions.checkNotNull(key, "get");
+
+    String value = null;
+    if (Files.exists(Paths.get(mFile))) {
+      try (InputStream is = new FileInputStream(mFile)) {
+        mProperties.load(is);
+        value = mProperties.getProperty(key);
+      } catch (Exception e) {
+        LOG.error("Failed to read file.", e);
+      }
+    }
+    return value;
+  }
+
+  /**
+   * persist a key-value to property file.
+   *
+   * @param key   the key to be persist into this file
+   * @param value the value corresponding to key
+   * @throws IOException I/O error if create or write file failed
+   */
+  @Override
+  public void set(String key, String value) throws IOException {
```

Comment:
I think we don't need a general `get()` and `set()` KV store. Let's start simple and just define `setClusterId()` and `getClusterId()`. If in the future we see more need, more methods will be added to this class, and then we can choose to abstract an interface.